packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -476,7 +484,19 @@ const DataGrid = injectIntl((props) => {
     if (!event.currentTarget.contains(event.relatedTarget)) {
       // Not triggered when swapping focus between children
       if (handleFocus.current) {
-        setFocusedRowCol(focusedRow, focusedCol, true);
+        // When modifying the number of rows, we set the focused cell to the last selected RowId and columnId
+        if (lastActiveRowId) {
```

Comment:
You have logic for the rows, but there does not seem to be logic to ensure that the column is valid.